### ISSUE TITLE
feat(if inline): add an `IfInline` node type

### ISSUE THREAD

**Issue by nokome** [stencila/stencila#2024]
### Summary

The schema has an [IfBlock](https://github.com/stencila/stencila/blob/main/docs/reference/schema/flow/if-block.md) for conditionally showing (and executing) block content. But there is currently no inline equivalent and some users have asked for it.

## Structure

The structure of an `IfInline` would follow an [`IfBlock`](https://github.com/stencila/stencila/blob/main/schema/IfBlock.yaml) closely but `clauses` would be an array of `IfInlineClause`s which itself would have `content` be an array of `Inline`s.

## Markdown syntax

Probably the thing about adding a new node type that needs the most thought is its Markdown representation. Currently the Markdown syntax for `IfBlock` is:

```md
::: if expr

This paragraph should be shown if the `expr` is truthy

:::
```

```md
::: if false

This first paragraph should NOT be shown

::: elif false

This second paragraph should NOT be shown

::: elif true

This third paragraph should be shown

::: elif true

This fourth paragraph should NOT be shown because above clause is

::: else 

This final paragraph should NOT be shown because above clause is

:::
```

More examples of the syntax in [`examples/nodes/if-block/if-block.md`](https://github.com/stencila/stencila/blob/main/examples/nodes/if-block/if-block.md).

Given that, I have two suggestions for syntax for `IfInline`. Both use curly braces, which we use for other inline extensions to Markdown syntax, but in different ways .

There is mention below of the need to specify the language to use in multi-lang docs. Given the parsing we do of code and knowledge of which variables are defined in which language kernels I expect that we can guess that reliably quite a lot of the time (ie. the user won't need to specify). However, it is good to have an escape hatch so that can be explicit.

### A. Most consistent with `IfBlock`

Uses the braces to encapsulate the expression for each clause and an `{end}` to signify the end of the `IfInline` (analogous to the the ending three semicolons of an `IfBlock`):

```md
The level of the dam was {if level > 10}very high{elif level > 5}high{else}normal{end}. 
```

When a language needed to be specified for the expression it could be done like this (`#py` chosen to avoid clash with other Markdown syntax, and because `#` is used for comments in Python and R):

```md
The level of the dam was {if level > 10 #py}very high{elif level > 5 #py}high{else}normal{end}. 
```

Also most consistent with templating languages such as [Handlebars](https://handlebarsjs.com/guide/block-helpers.html#conditionals) and [Jinja](https://jinja.palletsprojects.com/en/3.0.x/templates/#if).

### B. Most consistent with [CriticMarkup](https://fletcher.github.io/MultiMarkdown-6/syntax/critic.html) used for `InsertInline`, `ReplaceInline` etc.

Puts everything into braces, including the `content` and uses backticks to separate the expression from the content.

```md
The level of the dam was {if `level > 10` very high}{elif `level > 5` high}{else normal}. 
```

When a language needed to be specified:

```md
The level of the dam was {if `level > 10` #py very high}{elif `level > 5` #py high}{else normal}. 
```

Would appreciate your thoughts @GusEllerm @ignatiusm @brettc @edwardabraham.

**Comment by brettc** [comment]
First thoughts: Inline expressions tend to be look complex and a bit hard to parse so I would go for the simplest option (the first one). Keeping it in step with critic markup does not seem that important, as that is more about edits/deletes etc, rather than conditional templating.
